pallets/asset/src/lib.rs

```diff
@@ -809,13 +820,17 @@ decl_module! {
             // Granularity check
             ensure!(Self::check_granularity(&ticker, value), Error::<T>::InvalidGranularity);
             ensure!(<BalanceOf<T>>::contains_key(&ticker, &did), Error::<T>::NotAnOwner);
-            let burner_balance = Self::balance(&ticker, &did);
+            let (burner_balance, burner_def_balance) = Self::balance(&ticker, did);
             ensure!(burner_balance >= value, Error::<T>::InsufficientBalance);
+            ensure!(burner_def_balance >= value, Error::<T>::InsufficientDefaultPortfolioBalance);
 
             // Reduce sender's balance
             let updated_burner_balance = burner_balance
                 .checked_sub(&value)
-                .ok_or(Error::<T>::BalanceOverflow)?;
+                .ok_or(Error::<T>::BalanceUnderflow)?;
```

Comment:
This check is redundant since total balance will always be equal or more than the individual portfolio balance.